numpy/core/src/multiarray/common.h

```diff
@@ -169,6 +179,11 @@ check_and_adjust_axis(int *axis, int ndim)
     }
     return 0;
 }
+static NPY_INLINE int
+check_and_adjust_axis(int *axis, int ndim)
+{
+    return check_and_adjust_axis_msg(axis, ndim, Py_None);
```

Comment:
I guess my argument would be that if I manually inlined the contents of `check_and_adjust_axis_msg` into `normalize_axis_index`, you would not expect me to add any additional `Py_INCREF`s

I'm not sure what you mean there - can you elaborate on exactly why it is a bad idea?